core/lib/server/pub.ts

```diff
@@ -406,65 +415,22 @@ export const createPubRecursiveNew = async <Body extends CreatePubRequestBodyWit
 	const parentId = parent?.id ?? body.parentId;
 	const stageId = body.stageId;
 
-	const pubFieldsForCommunityObject = await getPubFields({
-		communityId,
-		includeRelations: true,
-	}).executeTakeFirst();
-
-	const pubFieldsForCommunity = Object.values(pubFieldsForCommunityObject?.fields ?? {});
-
-	if (!pubFieldsForCommunity?.length) {
-		throw new NotFoundError(`No pub fields found in community ${communityId}.`);
-	}
-
 	const values = body.values ?? {};
-	const filteredFields = pubFieldsForCommunity.filter((field) => {
-		const value = values[field.slug] ?? body.relatedPubs?.[field.slug];
-		return Boolean(value);
-	});
-
-	const normalizedValues = Object.fromEntries(
-		Object.entries(values).map(([slug, value]) =>
-			value != null && typeof value === "object" && "value" in value
-				? [slug, value.value]
-				: [slug, value]
-		)
-	);
-
-	const validationErrors = Object.values(
-		validatePubValuesBySchemaName({
-			fields: filteredFields,
-			values: normalizedValues,
-		})
+	const normalizedValues = Object.entries(values).flatMap(([slug, value]) =>
+		isRelatedPubInit(value)
+			? value.map((v) => ({ slug, value: v.value, relatedPubId: v.relatedPubId }))
+			: ([{ slug, value, relatedPubId: undefined }] as {
+					slug: string;
+					value: unknown;
+					relatedPubId: PubsId | undefined;
+				}[])
 	);
 
-	if (validationErrors.length) {
-		throw new Error(validationErrors.join(" "));
-	}
-
-	const valuesWithFieldIds = Object.entries(values).map(([slug, value]) => {
-		const field = filteredFields.find(
-			({ slug: slugInPubTypeFields }) => slug === slugInPubTypeFields
-		);
-		if (!field) {
-			throw new NotFoundError(`No pub field found for slug '${slug}'`);
-		}
-
-		const valueMaybeWithRelatedPubId =
-			value && typeof value === "object" && "value" in value
-				? { value: JSON.stringify(value.value), relatedPubId: value.relatedPubId as PubsId }
-				: { value: JSON.stringify(value) };
-
-		return {
-			id: field.id,
-			slug: field.slug,
-			...valueMaybeWithRelatedPubId,
```

Comment:
i also updated `createPubRecursiveNew` to use the new `validatePubValues` function